project-ideas/fulfillment-center-mgmt/readme.md

```diff
@@ -7,44 +7,64 @@
+
   *  The user can choose to [rejectOrderItem](rejectOrderItem.md) if for some reason orderItem cannot be fulfilled.
   *  The user can choose to [cancelOrderItem](cancelOrderItem.md) on request of the customer or Customer Service Rep (CSR).
 
-*   Step 2: User completes the [Packing](packShipment.md). Alternatively user can choose to edit autogenerated Shipment.
+### Step 2: 
+User completes the [Packing](packShipment.md). Alternatively user can choose to edit autogenerated Shipment.
   *  [rejectShipmentItem](rejectShipmentItem.md).
   *  Add [ShipmentPackage](createShipmentPackage.md)
   *  Move ShipmentItems between packages.
   *  After edits [packShipment](packShipment.md).
   *  [getShippingLabel](getShippingLabel.md).
-*   Step 3: User marks Shipment shipped.
+
+### Step 3:
+User marks Shipment shipped.
   * reprint shipping label and other documents
   * [voidShipmentLabel](voidShipmentLabel.md) and request it again.
 
 
-### [Shipment lifecycle](ShipmentStatusWorkflow.md)
+## [Shipment lifecycle](ShipmentStatusWorkflow.md)
 Shipment is created in SHIPMENT_INPUT, then SHIPMENT_APPROVED to SHIPMENT_PACKED and then SHIPMENT_SHIPPED
 
-**On SHIPMENT_APPROVED:**
-* On the SHIPMENT_APPROVED status event of Shipment lifecycle,  systems triggers the process to get Shipping Label (doRateShopping) from the logistics company.
+### On SHIPMENT_APPROVED:
+* Triggers the process to get Shipping Label (doRateShopping) from the logistics company
+  * If the facility is part of the `AUTO_SHIPPING_LABEL` facility group.
+
 * On successful execution of shipping label RateShopping, the RateShopping service updates the update ShipmentRouteSegment. Set `shipmentMethodTypeId`, `carrierPartyId`, `actualCost`, `carrierServiceStatusId` (SHRSCS_CONFIRMED). 
+
 * Once we have Shipping label, Shipment can be moved to SHIPMENT_PACKED status.
-* Ship (Handover in case of BOPIS) the Shipment moves Shipment status to S
+  * If `isTrackingRequired` is set to **N** in the ProductStoreShipmentMeth entity for the given Carrier Shipment Method, then the user can pack the shipment without a shipping label or tracking code.
 
-**On SHIPMENT_PACKED:**
+### On SHIPMENT_PACKED:
+* Ship (Handover in case of BOPIS) the Shipment moves Shipment status to Shipped
+* Fetch shipping label if auto shipping label is disabled at the facility and label is not already fetched.
 
-**On SHIPMENT_SHIPPED:**
+### On SHIPMENT_SHIPPED:
 * [createOrderShipmentInventoryIssuance](createOrderShipmentInventoryIssuance.md)
 * Order Item is marked Completed
 * Update Shopify order Fulfilled
 
 **Modify contents of the Shipment after it is already SHIPMENT_APPROVED or SHIPMENT_PACKED.**
+Why does the shipment have to be moved out of "approved" status to edit it's contents? For example fulfillment team may want to reject an item after begining pack pick process, in which case shipment will already be packed.
```

Comment:
Once Approved, we know that the packer will change the state of Shipment, i.e ShipmentItems and ShipmentPackageContent records will not be modified.

The Approved Shipment, is ready for getting Shipping Label because at this point we know the number of packages, their dimension and weight. 
We compute the package weight and save it in ShipmentPackage entity. 

If we make changes in content, we want to recompute the weight. 

This is why we want to move Shipment to Input status for any edits to its contents.